runtime/major_gc.c

```diff
@@ -749,12 +757,32 @@ static uintnat gc_slice_work(uintnat heap_words,
 
       return max2 (alloc_work, offheap_work);
     }
-  case GC_PACING_2025_FULL: /* TODO: separate this out. */
-    /* fall through */
-  case GC_PACING_2025_SIMPLE: {
+  case GC_PACING_2025: {
     /* Shiny new 2025 Doligez/Dolan pacing policy */
     double sweep_per_mark = (double)caml_percent_sweep_per_mark / 100.0;
     double space_overhead = (double)caml_percent_free / 100.0;
+    if (caml_gc_overhead_adjustment != 0) {
+      /* In theory, we should adjust according to the allocation rate.
+
+         However, this produces a dependence on real time, making the GC
+         nondeterministic even for single-threaded programs and doing weird
+         things for programs which pause in I/O for a long time.
+
+         Instead, we use minor heap allocation as a proxy for time, so instead
+         of measuring the allocation rate we measure the promotion rate.
+
+         The promotion rate is scaled so that the nominal promotion rate of 10%
```

Comment:
Perhaps measure; 10% is really a guess.

Reply:
Right, agreed.

But I'd point out that this is not a statement that "most programs have 10% promotion", but just a specification of what the space_overhead parameter does. (The 10% is effectively just a choice of units for space_overhead).

So if the average promotion rate across many programs turns out to be 7.35%, then I think we should still keep this at 10%, to keep the specification of what it does memorable.

(In other words, I agree we should measure more, but then we should pick this number to be whichever of 1%, 2%, 5%, 10%, 20% is closest to most programs)

Reply:
Yes, but we might want to change the default `space_overhead` accordingly.